app/models/mail_notifications/mail_notification.rb

```diff
@@ -23,12 +23,26 @@ class MailNotification < ActiveRecord::Base
 
   default_scope -> { order("mail_notifications.name") }
 
+  def self.new(args = nil, options = {})
+    if args[:name] == 'puppet_error_state'
```

Comment:
Kinda confused, why don't you just override `def initialize` and call `super(args)`?